graphql/server.go

```diff
@@ -36,6 +37,24 @@ type GraphqlLogger interface {
 	Error(ctx context.Context, err error, tags map[string]string)
 }
 
+type empty struct{}
+type semaphore chan empty
+
+// acquire n resources
```

Comment:
nit (and a little silly in a file that's as messy as it is): please end your comments with a period